src/Microsoft.AspNetCore.ResponseCompression/ResponseCompressionProvider.cs

```diff
@@ -49,14 +52,21 @@ public ResponseCompressionProvider(IServiceProvider services, IOptions<ResponseC
                 }
             }
 
-            var mimeTypes = options.Value.MimeTypes;
-            if (mimeTypes == null || !mimeTypes.Any())
+            _excludedMimeTypes = new HashSet<string>(
+                responseCompressionOptions.ExcludedMimeTypes ?? Enumerable.Empty<string>(),
+                StringComparer.OrdinalIgnoreCase
+            );
+
+            _mimeTypes = new HashSet<string>(
+                responseCompressionOptions.MimeTypes ?? Enumerable.Empty<string>(),
+                StringComparer.OrdinalIgnoreCase
+            );
+            if (_mimeTypes.Count == 0 && _excludedMimeTypes.Count == 0)
```

Comment:
Setting an exclusion wipes out the inclusion defaults? What's the point of exclusions if you don't have any inclusions?

You're missing a test case for this. E.g. No includes specified and one exclude specified. The default includes should still apply.